go-controller/pkg/util/ovs.go

```diff
@@ -839,7 +856,30 @@ func (t *NBTxn) Commit() (string, string, error) {
 	if len(t.txnArgs) == 0 {
 		return "", "", nil
 	}
-	allArgs := append(t.args, t.txnArgs...)
-	stdout, stderr, err := runOVNretry(runner.nbctlPath, t.env, allArgs...)
+
+	// assume a safe buffer of 10 args for a nbctl command
+	chunkSize := t.maxArgs - 10
+
+	stdout := bytes.NewBuffer([]byte{})
+	stderr := bytes.NewBuffer([]byte{})
+	var err error
+	// Chunk commands into acceptable sizes
+	// Warning if a transaction contains dependent chained transactions, like ACL commands,
+	// this could result in segmenting on those commands and then failing to apply them
+	for i := 0; i < len(t.txnArgs); i += chunkSize {
+		end := i + chunkSize
+
+		// necessary check to avoid slicing beyond
+		// slice capacity
+		if end > len(t.txnArgs) {
```

Comment:
I think we need to chunk at a -- boundary

Reply:
Yes